ICGE-Ui/src/main/java/de/unistuttgart/informatik/fius/icge/ui/internal/SwingEntityInspector.java

```
@@ -93,30 +102,130 @@ public void setName(final String name) {
     }
     
     /**
-     * Add a entry to the ui editor
+     * Update the entity inspector to reflect the new Values.
+     *
+     * @param entries
+     *     the new list of entries to display
+     */
+    public void updateEntityInspectorEntries(final EntityInspectorEntry[] entries) {
+        if (this.canUpdateValuesOnly(entries)) {
+            // only update values
+            SwingUtilities.invokeLater(() -> {
+                synchronized (this.updateLock) {
+                    // check preconditions
+                    List<Consumer<String>> valueUpdaters = this.uiValueUpdaters;
+                    if (valueUpdaters == null) return; // TODO throw exeptions here?
+                    if (valueUpdaters.size() != entries.length) return;
```

I think we should throw an exception in both cases. (We should catch and log it somewhere though)

---

These exceptions could only happen if the update happens while the inspector is still updating. If I understand SwingUtilities.invokeLater correctly such a race condition should not happen. It is also synchronized for good measure and these checks are just to be extra sure that there will never be any exception presented to the user because the entity inspector had a bug. So throwing an exception here may not be for the best as the worst outcome is that the displayed data is from the last tick.